### ISSUE TITLE
Query limit not overriding role row limit

### ISSUE THREAD

**Issue by matheus1lva** [hasura/graphql-engine#8131]
### Version Information

Server Version: hasura/graphql-engine:v2.0.10.cli-migrations-v3

### Environment

docker on ubuntu 20

### What is the expected behaviour?

Having a row limit on a role set to 20, for eg.

And when asking for data:

```
{
 users(limit: 50) {
   id
 }
}
```

It should return 50 results instead of 20
### Keywords

row limit, pagination with row limit, pagination override.

### What is the current behaviour?

It returns only 20 rows, instead of 50. Ive tried both the relay endpoint and the non relay one.

**Comment by matheus1lva** [comment]
To be clear, this only happens with relay, if you try to grab more results past the 39th cursor, it shows as null and does not show the `hasNextPage` info on `pageInfo`

![image](https://user-images.githubusercontent.com/831308/152401052-49887773-ee43-4b9f-8950-8c1da9fa0a9f.png)

**Comment by arjunyel** [comment]
Is this happening on version 2.2.0?

**Comment by matheus1lva** [comment]
Yes, the pageInfo still sends `hasNextPage` as false!

**Comment by BenoitRanque** [comment]
Tested this, able to reproduce. It looks like the roles' row limit is being treated as the total number of rows, which is incorrect.
To reproduce this:
- set a row limit on a role
- query a table using the relay api, set the `first` argument equal or greater than the row limit
- the hasNextPage value will be false.

Do note: the row limit is _intended_ to override the desired amount of rows specified using `first`.
However users should still be able to paginate beyond the maximum amount of rows they are allowed to query at once.

Therefore your query returning 20 rows is expected behavior. However you should be able to paginate to page 2 and 3, with 20 and 10 rows each respectively (assuming there are a total of 50 selectable rows)

**Comment by joanrodriguez** [comment]
Yes @BenoitRanque, setting a limit on the table breaks pagination.

**Comment by PedroBern** [comment]
I can reproduce this issue. It works fine if I leave the row limit blank for a given role (which is not ideal)
